File: spring-backend/src/main/java/sg/edu/nus/iss/wellness/dto/AuthDtos.java
package sg.edu.nus.iss.wellness.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * DTOs for authentication endpoints.
 *
 * @author SA62 Team
 */
public final class AuthDtos {
    private AuthDtos() {
    }

    public record RegisterRequest(
            @NotBlank String displayName,
            @NotBlank @Email String email,
            @NotBlank @Size(min = 8) String password
    ) {
    }

    public record LoginRequest(
            @NotBlank @Email String email,
            @NotBlank String password
    ) {
    }

    public record UserResponse(Long id, String displayName, String email) {
    }

    public record LoginResponse(String token, String tokenType, long expiresInSeconds, UserResponse user) {
    }
}

File: dotnet-backend/Dtos/AuthDtos.cs
namespace Wellness.Backup.Api.Dtos;

/// <summary>
/// DTOs for authentication endpoints mirrored from Spring Boot.
/// </summary>
/// <remarks>@author SA62 Team</remarks>
public sealed record RegisterRequest(string? DisplayName, string? Email, string? Password);

public sealed record LoginRequest(string? Email, string? Password);

public sealed record UserResponse(long Id, string DisplayName, string Email);

public sealed record LoginResponse(string Token, string TokenType, long ExpiresInSeconds, UserResponse User);
